src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBranch.java

```diff
@@ -285,4 +295,42 @@ private static boolean isZeroImpedanceBranch(PiModel piModel, boolean dc, double
             return piModel.getZ() < lowImpedanceThreshold;
         }
     }
+
+    private void updateZeroImpedanceNetworks(boolean disabled, boolean dc) {
+        if (isZeroImpedance(dc)) {
+            LfZeroImpedanceNetwork zn1 = bus1.getZeroImpedanceNetwork(dc);
+            LfZeroImpedanceNetwork zn2 = bus2.getZeroImpedanceNetwork(dc);
+            if (zn1 != null && zn2 != null) {
+                if (disabled) {
+                    if (zn1 == zn2) {
+                        // zero impedance network split (maybe)
+                        zn1.removeBranchAndTryToSplit(this);
+                    } else {
+                        throw new IllegalStateException("Should not happen");
+                    }
+                } else {
+                    if (zn1 != zn2) {
+                        // zero impedance network merge
+                        LfZeroImpedanceNetwork.addBranchAndMerge(zn1, zn2, this);
+                    } else {
+                        // we need to add the branch again to zero impedance graph and update spanning tree as
```

Comment:
the branch needs to be added, sure, but the previous spanning tree is still **a** spanning tree. It might not be a minimal spanning tree, but is that a problem? If the minimal part of the spanning tree is essential, maybe we should emphasize it in the namings.

Reply:
Not clear anymore for me what was the original issue we tried to solve with this. This line of code is not covered by a unit test but we absolutely need to do it. @annetill help would be welcome to try to create a use case from the original real data to cover this case.

Reply:
Are we talking about the Sunday's fix?

Reply:
Yes

Reply:
@flo-dup looking at new TU added @annetill I think this case covers an addition to the zero impedance graph (so a new bus that has just joined another zero impedanc graph). So it was not anymore a spanning tree of the graph as a vertex was missing.

Reply:
@flo-dup I think you are right, the unit test added by @annetill still pass by just adding the edge but not recomputing the spanning tree